test/jdk/java/lang/reflect/code/TestStringConcatTransform.java

```diff
@@ -0,0 +1,278 @@
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+/*
+ * @test
+ * @run testng TestStringConcatTransform
+ */
+
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.NoInjection;
+import org.testng.annotations.Test;
+import java.lang.reflect.code.transformations.LinearConcatTransform;
+
+import java.lang.invoke.MethodHandle;
+import java.lang.invoke.MethodHandles;
+import java.lang.reflect.Method;
+import java.lang.reflect.code.Op;
+import java.lang.reflect.code.analysis.SSA;
+import java.lang.reflect.code.bytecode.BytecodeGenerator;
+import java.lang.reflect.code.interpreter.Interpreter;
+import java.lang.reflect.code.op.CoreOp;
+import java.lang.runtime.CodeReflection;
+import java.util.Arrays;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+
+public class TestStringConcatTransform {
+
+    static final String TESTSTR = "TESTING STRING";
+
+    static final Map<Class<?>, Object> valMap;
+
+    static {
+        valMap = new HashMap<>();
+        valMap.put(byte.class, (byte) 42);
+        valMap.put(short.class, (short) 42);
+        valMap.put(int.class, 42);
+        valMap.put(long.class, (long) 42);
+        valMap.put(float.class, 42f);
+        valMap.put(double.class, 42d);
+        valMap.put(char.class, 'z');
+        valMap.put(boolean.class, false);
+
+        valMap.put(Byte.class, (byte) 42);
+        valMap.put(Short.class, (short) 42);
+        valMap.put(Integer.class, 42);
+        valMap.put(Long.class, (long) 42);
+        valMap.put(Float.class, 42f);
+        valMap.put(Double.class, 42d);
+        valMap.put(Character.class, 'z');
+        valMap.put(Boolean.class, false);
+
+        valMap.put(Object.class, new Object() {
+            @Override
+            public String toString() {
+                return "I'm a test string.";
+            }
+        });
+        valMap.put(TestObject.class, new TestObject());
+        valMap.put(String.class, TESTSTR);
+    }
+
+    public static final class TestObject {
+        TestObject() {
+        }
+
+        @Override
+        public String toString() {
+            return "TestObject String";
+        }
+    }
+
+    @Test(dataProvider = "getClassMethods")
+    public void testModelTransform(@NoInjection Method method) {
+        CoreOp.FuncOp model = method.getCodeModel().orElseThrow();
+        CoreOp.FuncOp f_transformed = model.transform(new LinearConcatTransform.ConcatTransform());
+        Object[] args = prepArgs(method);
+
+        System.out.println("Final Transformed");
+        f_transformed.writeTo(System.out);
+
+        var interpreted = Interpreter.invoke(model, args);
+        var transformed_interpreted = Interpreter.invoke(f_transformed, args);
+
+        Assert.assertEquals(interpreted, transformed_interpreted);
+
+    }
+
+    @Test(dataProvider = "getClassMethods")
+    public void testSSAModelTransform(@NoInjection Method method) {
```

Comment:
Also transform model as in `model.transform(new LinearConcatTransform.ConcatTransform())` i.e. before SSA too.